File: backend/internal/models/verification_fields.go
package models

type VerificationField struct {
	Id        string `json:"id" db:"id"`
	SectionId string `json:"sectionId" db:"section_id"`
	Field     string `json:"field" db:"field"`
	Label     string `json:"label" db:"label"`
	Type      string `json:"type" db:"type"`
	Position  int    `json:"position" db:"position"`
	Group     string `json:"group" db:"group"`
}

type GetVerFieldsDTO struct {
	SectionId string `json:"sectionId" db:"section_id"`
	Group     string `json:"group" db:"group"`
}

type VerificationFieldDTO struct {
	Id        string `json:"id" db:"id"`
	SectionId string `json:"sectionId" db:"section_id"`
	Field     string `json:"field" db:"field"`
	Label     string `json:"label" db:"label"`
	Type      string `json:"type" db:"type"`
	Position  int    `json:"position" db:"position"`
	Group     string `json:"group" db:"group"`
}

type DeleteVerFieldDTO struct {
	Id string `json:"id" db:"id"`
}

File: frontend/src/features/table/modules/verification/modules/types/verificationFields.ts
export interface IVerificationField {
	id: string
	sectionId: string
	field: string
	label: string
	type: string
	position: number
	width: number
}

export interface IVerificationFieldDTO {
	id: string
	sectionId: string
	field: string
	label: string
	type: string
	position: number
	group: string
	width?: number
}

export interface IVerificationFieldForm {
	id: string
	sectionId: string
	field: string
	label: string
	type: string
	position: number
	width: number
	group: string
	status?: 'new' | 'updated' | 'deleted' | 'moved' | 'none'
}
